### PR TITLE
Mjmadhu playwrighttesting microsoft.azure playwright service 2023 10 01 preview new

### PR DESCRIPTION
**Pull request by mjmadhu**
# ARM (Control Plane) API Specification Update Pull Request 

> Tip: overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] [ARM review queue] (for **merge** queues, see [2])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

</details>

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the Diagram Step 2, "ARM API changes Review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by mjmadhu** [comment]
The base commit is already added, hence removing this tag.

File: specification/playwrighttesting/PlaywrightTesting.Management/main.tsp [hunk 1]
```diff
@@ -13,28 +13,28 @@ using Azure.Core;
 @armProviderNamespace("Microsoft.AzurePlaywrightService")
 @service({
   title: "Microsoft AzurePlaywrightService Management API",
-  version: "2023-10-01-preview",
+  version: "2024-02-01",
 })
-@doc("Microsoft.AzurePlaywrightService Resource Provider management API.")
+@doc("Microsoft.AzurePlaywrightService Resource Provider Management API.")
 @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
```
Comment (raosuhas) [inline specification/playwrighttesting/PlaywrightTesting.Management/main.tsp:19]: @armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v5)

**Comment by raosuhas** [comment]
Could you please fix the newly introduced LintDiff errors : https://github.com/Azure/azure-rest-api-specs/pull/27132/checks?check_run_id=20718479251

File: specification/playwrighttesting/PlaywrightTesting.Management/main.tsp (hunk 1 above)
Comment (mjmadhu) [reply specification/playwrighttesting/PlaywrightTesting.Management/main.tsp:19]: done

**Comment by mjmadhu** [comment]
The remaining [Swagger LintDiff Errors](https://github.com/Azure/azure-rest-api-specs/pull/27132/checks?check_run_id=20760919249) are occurring from the TypeSpec generated code. Please let us know what changes need to be done if fixes are available for them.

[ResourceNameRestriction](https://github.com/Azure/azure-openapi-validator/blob/main/docs/resource-name-restriction.md) - It's throwing the following error as it's enum type -
error decorator-wrong-target: Cannot apply @pattern decorator to type it is not a string
> 109 |   @pattern("^[a-zA-Z]{1}[a-zA-Z0-9]{2,127}$")

[ProvisioningStateMustBeReadOnly](https://github.com/Azure/azure-openapi-validator/blob/main/docs/provisioning-state-must-be-read-only.md) - Already defined with @visibility("read") in the tsp file

[DeleteResponseCodes](https://github.com/Azure/azure-openapi-validator/blob/main/docs/delete-response-codes.md) - Default operations are generated from the TrackedResourceOperations as verfied from the following tsp code - 
@armResourceOperations
interface Accounts
  extends TrackedResourceOperations<Account, AccountProperties> {
  @doc("Adds check global name availability operation, normally used if a resource name must be globally unique.")
  checkNameAvailability is checkGlobalNameAvailability;
}

[ValidFormats](https://github.com/Azure/azure-openapi-validator/blob/main/docs/valid-formats.md) - Can't find any other format than string and int32 in the documentation to support decimal/float types.

[ParameterNotUsingCommonTypes](https://github.com/Azure/azure-openapi-validator/blob/main/docs/parameter-not-using-common-types.md) - The current [@locationResource](https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/reference/decorators#@Azure.ResourceManager.locationResource) does not comply with the signature of ARM location based resources like [usages ](https://armwiki.azurewebsites.net/api_contracts/UsagesAPIContract.html#request) signature mentioned here, hence created custom locations resource as discussed on the Teams Thread.

**Comment by raosuhas** [comment]
> The remaining [Swagger LintDiff Errors](https://github.com/Azure/azure-rest-api-specs/pull/27132/checks?check_run_id=20760919249) are occurring from the TypeSpec generated code. Please let us know what changes need to be done if fixes are available for them.
> 
> [ResourceNameRestriction](https://github.com/Azure/azure-openapi-validator/blob/main/docs/resource-name-restriction.md) - It's throwing the following error as it's enum type - error decorator-wrong-target: Cannot apply @pattern decorator to type it is not a string
> 
> > 109 |   @pattern("^[a-zA-Z]{1}[a-zA-Z0-9]{2,127}$")
> 
> [ProvisioningStateMustBeReadOnly](https://github.com/Azure/azure-openapi-validator/blob/main/docs/provisioning-state-must-be-read-only.md) - Already defined with @visibility("read") in the tsp file
> 
> [DeleteResponseCodes](https://github.com/Azure/azure-openapi-validator/blob/main/docs/delete-response-codes.md) - Default operations are generated from the TrackedResourceOperations as verfied from the following tsp code - @armResourceOperations interface Accounts extends TrackedResourceOperations<Account, AccountProperties> { @doc("Adds check global name availability operation, normally used if a resource name must be globally unique.") checkNameAvailability is checkGlobalNameAvailability; }
> 
> [ValidFormats](https://github.com/Azure/azure-openapi-validator/blob/main/docs/valid-formats.md) - Can't find any other format than string and int32 in the documentation to support decimal/float types.
> 
> [ParameterNotUsingCommonTypes](https://github.com/Azure/azure-openapi-validator/blob/main/docs/parameter-not-using-common-types.md) - The current [@locationResource](https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/reference/decorators#@Azure.ResourceManager.locationResource) does not comply with the signature of ARM location based resources like [usages ](https://armwiki.azurewebsites.net/api_contracts/UsagesAPIContract.html#request) signature mentioned here, hence created custom locations resource as discussed on the Teams Thread.

Ack : 
We should not need

> The remaining [Swagger LintDiff Errors](https://github.com/Azure/azure-rest-api-specs/pull/27132/checks?check_run_id=20760919249) are occurring from the TypeSpec generated code. Please let us know what changes need to be done if fixes are available for them.
> 
> [ResourceNameRestriction](https://github.com/Azure/azure-openapi-validator/blob/main/docs/resource-name-restriction.md) - It's throwing the following error as it's enum type - error decorator-wrong-target: Cannot apply @pattern decorator to type it is not a string
> 
> > 109 |   @pattern("^[a-zA-Z]{1}[a-zA-Z0-9]{2,127}$")
> 
> [ProvisioningStateMustBeReadOnly](https://github.com/Azure/azure-openapi-validator/blob/main/docs/provisioning-state-must-be-read-only.md) - Already defined with @visibility("read") in the tsp file
> 
> [DeleteResponseCodes](https://github.com/Azure/azure-openapi-validator/blob/main/docs/delete-response-codes.md) - Default operations are generated from the TrackedResourceOperations as verfied from the following tsp code - @armResourceOperations interface Accounts extends TrackedResourceOperations<Account, AccountProperties> { @doc("Adds check global name availability operation, normally used if a resource name must be globally unique.") checkNameAvailability is checkGlobalNameAvailability; }
> 
> [ValidFormats](https://github.com/Azure/azure-openapi-validator/blob/main/docs/valid-formats.md) - Can't find any other format than string and int32 in the documentation to support decimal/float types.
> 
> [ParameterNotUsingCommonTypes](https://github.com/Azure/azure-openapi-validator/blob/main/docs/parameter-not-using-common-types.md) - The current [@locationResource](https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/reference/decorators#@Azure.ResourceManager.locationResource) does not comply with the signature of ARM location based resources like [usages ](https://armwiki.azurewebsites.net/api_contracts/UsagesAPIContract.html#request) signature mentioned here, hence created custom locations resource as discussed on the Teams Thread.



Please add a suppression for : "ResourceNameRestriction" ,  "ProvisioningStateMustBeReadOnly" and "DeleteResponseCode" for the specific line where it is getting raised. Seems like bug in the tooling

For ValidTypes have you already tried float : https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#data-types ? or else you might have to use string since it wouldnt get recognized otherwise.

**Comment by raosuhas** [comment]
Signing off on this. The suppressions do not seem to be working. We might have to fix them before merge can complete though

**Comment by raosuhas** [comment]
> /pr RequestMerge

the suppressions you added do not seem to be working. Can you please go through the trouble shooting section in this doc to ensure that the suppressions are working as expected : https://eng.ms/docs/products/azure-developer-experience/design/specs-pr-guides/pr-suppressions 

suppressions must be for the correct file and package version not at the top level like you have added : 

I am removing the PR merge for now since we need to fix the suppressions before merge.

**Comment by mjmadhu** [comment]
@raosuhas Suppressions fixed now. Please help review and merge.

File: specification/playwrighttesting/resource-manager/readme.md [hunk 2]
```diff
@@ -29,7 +29,26 @@ title: PlaywrightTestingClient
...
+### Tag: package-2024-02-01
+
+These settings apply only when `--tag=package-2024-02-01` is specified on the command line.
+
+```yaml $(tag) == 'package-2024-02-01'
+input-file:
+  - Microsoft.AzurePlaywrightService/stable/2024-02-01/playwrighttesting.json
+suppressions:
+  - code: ResourceNameRestriction
+    from: playwrighttesting.json
```
Comment (visingla-ms) [inline specification/playwrighttesting/resource-manager/readme.md:44]: Please provide full path where this suppression is being applied. Whole file name is not correct.

File: specification/playwrighttesting/resource-manager/readme.md [hunk 3]
```diff
@@ -29,7 +29,26 @@ title: PlaywrightTestingClient
...
+    reason: We have enums defined for quota resource names which doesn't support string pattern validation.
+  - code: ProvisioningStateMustBeReadOnly
+    from: playwrighttesting.json
```
Comment (visingla-ms) [inline specification/playwrighttesting/resource-manager/readme.md:47]: Add readOnly true inside property definition, not where it is referenced. That will fix the linter issue and you can remove suppression for this.

File: specification/playwrighttesting/resource-manager/readme.md [hunk 4]
```diff
@@ -29,7 +29,26 @@ title: PlaywrightTestingClient
...
+    reason: Seems like a tool bug, as the visibility is set to read only in the TypeSpec already.
+  - code: DeleteResponseCodes
+    from: playwrighttesting.json
```
Comment (visingla-ms) [inline specification/playwrighttesting/resource-manager/readme.md:50]: Please provide full path where this suppression is being applied. Whole file name is not correct.

File: specification/playwrighttesting/resource-manager/readme.md (hunk 3 above)
Comment (mjmadhu) [reply specification/playwrighttesting/resource-manager/readme.md:47]: @visingla-ms the swagger json is being generated from TypeSpec file. We've followed the syntax to put visibility there. Changing it's place gives this error -

image

File: specification/playwrighttesting/resource-manager/readme.md (hunk 4 above)
Comment (mjmadhu) [reply specification/playwrighttesting/resource-manager/readme.md:53]: Done

File: specification/playwrighttesting/resource-manager/readme.md (hunk 2 above)
Comment (mjmadhu) [reply specification/playwrighttesting/resource-manager/readme.md:44]: Done

**Comment by mjmadhu** [comment]
Paths doesn't seem to be working with ResourceNameRestriction Suppression as observed in existing usage from the TSG as well -
https://github.com/Azure/azure-rest-api-specs/blob/0c2e0ff7acea901abb0d6714385546788c10d878/specification/desktopvirtualization/resource-manager/readme.md?plain=1#L40

**Comment by mjmadhu** [comment]
Please help us with review and merge before the Chinese holiday season next week.

**Review by visingla-ms — APPROVED** [review]

**Pull request merged**